Content/Console/build.fsx

```diff
@@ -161,12 +226,40 @@ let clean _ =
     ]
     |> Seq.iter Shell.rm
 
+let updateChangelog ctx =
+    let description, unreleasedChanges =
+        match changelog.Unreleased with
+        | None -> None, []
+        | Some u -> u.Description, u.Changes
+    let verStr = ctx |> getVersionNumber "RELEASE_VERSION"
+    let newVersion = SemVer.parse verStr
+    if changelog.Entries |> List.exists (fun entry -> entry.SemVer = newVersion) then
+        let entry = changelog.Entries |> List.find (fun entry -> entry.SemVer = newVersion)
+        Trace.traceErrorfn "Version %s already exists in %s, released on %s" verStr changelogFilename (if entry.Date.IsSome then entry.Date.Value.ToString("yyyy-MM-dd") else "(no date specified)")
+        failwith "Can't release with a duplicate version number"
+    if changelog.Entries |> List.exists (fun entry -> entry.SemVer > newVersion) then
+        let entry = changelog.Entries |> List.find (fun entry -> entry.SemVer > newVersion)
+        Trace.traceErrorfn "You're trying to release version %s, but a later version %s already exists, released on %s" verStr entry.SemVer.AsString (if entry.Date.IsSome then entry.Date.Value.ToString("yyyy-MM-dd") else "(no date specified)")
+        failwith "Can't release with a version number older than an existing release"
+    let versionTuple version = (version.Major, version.Minor, version.Patch)
+    let prereleaseEntries = changelog.Entries |> List.filter (fun entry -> entry.SemVer.PreRelease.IsSome && versionTuple entry.SemVer = versionTuple newVersion)
+    let prereleaseChanges = prereleaseEntries |> List.collect (fun entry -> entry.Changes |> List.filter (not << isEmptyChange))
+    let assemblyVersion, nugetVersion = Changelog.parseVersions newVersion.AsString
+    linkReferenceForLatestEntry <- mkLinkReference newVersion changelog
+    let newEntry = Changelog.ChangelogEntry.New(assemblyVersion.Value, nugetVersion.Value, Some System.DateTime.Today, description, unreleasedChanges @ prereleaseChanges, false)
+    let newChangelog = Changelog.Changelog.New(changelog.Header, changelog.Description, None, newEntry :: changelog.Entries)
+    latestEntry <- newEntry
+    newChangelog
+    |> Changelog.save changelogFilename
+    // Changelog.save doesn't write a final newline, so we add one when writing out the new link reference
+    sprintf "\n%s\n" linkReferenceForLatestEntry |> File.writeString true changelogFilename
```

Comment:
When generating these new links, the order of the links is opposite of how [KeepAChangelog](https://keepachangelog.com/en/1.0.0/) seems to do it. 

```
[0.1.0]: https://github.com/user/MyCoolNewLib.git/releases/tag/v0.1.0
[0.2.0-beta001]: https://github.com/TheAngryByrd/miniscaffold-docs-test-3/compare/v0.1.0...v0.2.0-beta001
[0.2.0-beta002]: https://github.com/TheAngryByrd/miniscaffold-docs-test-3/compare/v0.1.0...v0.2.0-beta002
[0.2.0]: https://github.com/TheAngryByrd/miniscaffold-docs-test-3/compare/v0.1.0...v0.2.0
```

[Full file](https://raw.githubusercontent.com/TheAngryByrd/miniscaffold-docs-test-3/master/CHANGELOG.md)

Reply:
It's quite a bit simpler to just append a line, but I've thought of a way to do it in the Keep A Changelog order (so the link reference targets appear in the file in the same order as the corresponding changelog entries) and still keep the code relatively simple. It's going to be more complicated than the rest of these fixes, so I think I'll put that in a different commit from the other fixes.

Reply:
I’d say it’s not really worth it if it’s a lot of work  just wanted to point it it out.

Reply:
I’d say it’s not really worth it if it’s a lot of work  just wanted to point it it out.

Reply:
It's not going to be that much work, and it will also allow me to update the `[Unreleased]` link target at the same time. I'm going to do it.

Reply:
Done in 90897dc.